.gitlab-ci.yml

```diff
@@ -215,70 +213,67 @@ publish:linux:snap:amd64:          &publish_snap
   variables:
     BUILD_ARCH:                    amd64
   dependencies:
-    - package:linux:snap:amd64
+    - package-linux-snap-amd64
   script:
     - scripts/gitlab/publish-snap.sh
   tags:
     - rust-stable
 
-publish:linux:snap:i386:
+publish-linux-snap-i386:
   <<:                              *publish_snap
   before_script:
     - *determine_version
-    #- sleep 60
   variables:
     BUILD_ARCH:                    i386
   dependencies:
-    - package:linux:snap:i386
+    - package-linux-snap-i386
 
-publish:linux:snap:arm64:
+publish-linux-snap-arm64:
   <<:                              *publish_snap
   before_script:
     - *determine_version
-    #- sleep 360
   variables:
     BUILD_ARCH:                    arm64
   dependencies:
-    - package:linux:snap:arm64
+    - package-linux-snap-arm64
 
-publish:linux:snap:armhf:
+publish-linux-snap-armhf:
   <<:                              *publish_snap
   before_script:
     - *determine_version
-    #- sleep 520
   variables:
     BUILD_ARCH:                    armhf
   dependencies:
-    - package:linux:snap:armhf
+    - package-linux-snap-armhf
 
-publish:docker:parity:amd64: &publish_docker
+publish-docker-parity-amd64: &publish_docker
   stage:                           publish
   only:                            *publishable_branches
   cache: {}
   dependencies:
-    - build:linux:ubuntu:amd64
+    - build-linux-ubuntu-amd64
   tags:
     - shell
   allow_failure:                   true
   script:
     - scripts/gitlab/publish-docker.sh parity
 
-publish:docker:parity-evm:amd64:
+publish-docker-parityevm-amd64:
   <<:                              *publish_docker
   script:
     - scripts/gitlab/publish-docker.sh parity-evm
 
-publish:github:s3:
+publish-github-and-s3:
```

Comment:
In my ideal world we would split this into two separate steps.
But let's merge it as it is for now.